src/content/en/tools/service-worker-libraries/index.markdown

```diff
@@ -0,0 +1,138 @@
+---
+layout: shared/wide
+description: "Service Worker Libraries."
+title: "Service Worker Libraries"
+order: 6
+feedName: sw-toolbox Releases
+feedURL: https://github.com/googlechrome/sw-toolbox/releases.atom
+---
+
+<div class="wf-subheading">
+  <div class="page-content mdl-typography--text-center mdl-grid">
+    <div class="mdl-cell mdl-cell--1-col"></div>
+    <div class="mdl-cell mdl-cell--10-col">
+      <h3>Service workers, without the work.</h3>
+      <p>
+        Free your web app from the network without writing boilerplate service worker code.
+      </p>
+      <div class="page-content mdl-grid">
+        <div class="mdl-cell mdl-cell--6-col">
+          <a class="mdl-button mdl-js-button mdl-button--raised" href="https://github.com/GoogleChrome/sw-precache#install">Get sw-precache</a>
+        </div>
+        <div class="mdl-cell mdl-cell--6-col">
+          <a class="mdl-button mdl-js-button mdl-button--raised" href="https://github.com/GoogleChrome/sw-toolbox#installing-service-worker-toolbox">Get sw-toolbox</a>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+
+<div class="page-content">
+  <h2>Ship your shell with <code>sw-precache</code></h2>
+
+  <div class="mdl-grid mdl-typography--text-center">
+    <div class="mdl-cell mdl-cell--4-col">
+      <div class="icon">
+        {%include svgs/design-and-ui.svg %}
+      </div>
+      <h3 class="mdl-typography--title">Precache Your Shell</h3>
+      <p>
+         Your web app's shell—its core HTML, JavaScript, and CSS—can be precached when a
+         user visits your page.
+      </p>
+    </div>
+
+    <div class="mdl-cell mdl-cell--4-col">
+      <div class="icon">
+        {%include svgs/build-chain.svg %}
+      </div>
+      <h3 class="mdl-typography--title">Build-time Integration</h3>
+      <p>
+        Drop it into your existing build process:
+        <a href="https://github.com/GoogleChrome/sw-precache/blob/master/demo/gulpfile.js">Gulp</a>,
+        <a href="https://github.com/GoogleChrome/sw-precache/blob/master/demo/Gruntfile.js">Grunt</a>, or
+        <a href="https://github.com/GoogleChrome/sw-precache#command-line-interface">command-line</a>.
+      </p>
+    </div>
+
+    <div class="mdl-cell mdl-cell--4-col">
+      <div class="icon">
+        {%include svgs/live-reload.svg %}
+      </div>
+      <h3 class="mdl-typography--title">Stay Fresh</h3>
+      <p>
+        Changes in your build update the service worker script. Users get updates without having to
+        manually version your content or caches.
+      </p>
+    </div>
+
+    <div class="mdl-cell mdl-cell--4-col">
+      <div class="icon">
+        {%include svgs/offline.svg %}
+      </div>
+      <h3 class="mdl-typography--title">No Network, No Problem</h3>
+      <p>
+        Your static resources are served
+        <a href="https://jakearchibald.com/2014/offline-cookbook/#cache-falling-back-to-network">cache-first</a>,
+        whether or not there's a network available.
```

Comment:
Since there are other caching patterns, perhaps the pattern used by this one shouldn't be buried in the text.